application/clicommands/CleanupCommand.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Icinga\Module\Businessprocess\Clicommands;
+
+use Icinga\Application\Logger;
+use Icinga\Application\Modules\Module;
+use Icinga\Cli\Command;
+use Icinga\Module\Businessprocess\Modification\NodeModifyAction;
+use Icinga\Module\Businessprocess\Modification\NodeRemoveAction;
+use Icinga\Module\Businessprocess\ProvidedHook\Icingadb\IcingadbSupport;
+use Icinga\Module\Businessprocess\State\IcingaDbState;
+use Icinga\Module\Businessprocess\State\MonitoringState;
+use Icinga\Module\Businessprocess\Storage\LegacyStorage;
+
+class CleanupCommand extends Command
+{
+    /**
+     * @var LegacyStorage
+     */
+    protected $storage;
+
+    protected $defaultActionName = 'cleanup';
+
+    public function init()
+    {
+        $this->storage = LegacyStorage::getInstance();
+    }
+
+    /**
+     * Cleanup all missing monitoring nodes from the specified config name
+     * If no config name is specified, the missing nodes are cleaned from all available configs
+     *
+     * USAGE
+     *
+     * icingacli businessprocess cleanup [<config-name>]
+     *
+     * OPTIONS
+     *
+     *   <config-name>
+     */
+    public function cleanupAction()
+    {
+        $configNames = (array) $this->params->shift() ?: $this->storage->listAllProcessNames();
+        $missingChildren = $this->listAllMissingChildren($configNames);
+        $removedNodes = [];
+
+        foreach ($missingChildren as $nodeName) {
+            foreach ($configNames as $configName) {
+                if ($this->storage->hasProcess($configName)) {
+                    $bp = $this->storage->loadProcess($configName);
```

Comment:
`listAllMissingChildren` did all of this already. Cache it. Use it. Or just merge the two methods, or use a generator. Just be more efficient please.